modules/claims_api/app/models/claims_api/power_of_attorney.rb

```diff
@@ -16,6 +17,70 @@ class PowerOfAttorney < ApplicationRecord
     before_validation :set_md5
     validates :md5, uniqueness: true
 
+    def sign_pdf
+      signatures = convert_signatures_to_images
+      page_1_path = insert_signatures(1, signatures[:veteran], signatures[:representative])
+      page_2_path = insert_signatures(2, signatures[:veteran], signatures[:representative])
+      { page1: page_1_path, page2: page_2_path }
+    end
+
+    def convert_signatures_to_images
+      {
+        veteran: convert_base64_data_to_image('veteran'),
+        representative: convert_base64_data_to_image('representative')
+      }
+    end
+
+    def convert_base64_data_to_image(signature)
+      path = "tmp/#{signature}_#{id}_signature_b64.png"
+
+      File.open(path, 'wb') do |f|
+        f.write(Base64.decode64(form_data.dig('signatures', signature)))
+      end
+      path
+    end
```

Comment:
I think I've already mentioned this, but I'm worried that two requests using the same id could try to be writing to the same file. We don't need `#{SecureRandom.hex}` in this filename?

Reply:
No this wont be an issue. Actually based on the current way ITF works, if the headers are not changed, this process would never run a second time for a veteran. When we do change that, this method gives us a reliable name without having to persist another field in the database. In your case above, we would want a new file to be written and used.